scripts/check_architecture_diagrams.py

```
@@ -0,0 +1,137 @@
+BINARY_SUFFIXES = frozenset({".gif", ".ico", ".jpg", ".jpeg", ".parquet", ".pdf", ".png", ".svg"})
+
+
+def _normalise_name(value: str) -> str:
+    """Return a filename fragment suitable for convention comparisons."""
+    return re.sub(r"[^a-z0-9]+", "", value.lower())
+
+
+def tracked_files(root: Path) -> list[Path]:
+    """Return repository-relative files tracked by Git."""
+    result = subprocess.run(
+        ["git", "ls-files", "-z"],
```

Ruff (0.16.1)

[error] Starting a process with a partial executable path

(S607)